File: Assets/Bundles/Shared/Resources/TS/UserInput/Drivers/Signals/KeySignal.lua
-- Compiled with unity-ts v2.1.0-75
local Cancellable = require("Shared/TS/Util/Cancellable").Cancellable
local KeySignal
do
	local super = Cancellable
	KeySignal = setmetatable({}, {
		__tostring = function()
			return "KeySignal"
		end,
		__index = super,
	})
	KeySignal.__index = KeySignal
	function KeySignal.new(...)
		local self = setmetatable({}, KeySignal)
		return self:constructor(...) or self
	end
	function KeySignal:constructor(Key)
		super.constructor(self)
		self.Key = Key
	end
end
return {
	KeySignal = KeySignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/UserInput/Drivers/Signals/KeySignal.ts
import { Cancellable } from "Shared/Util/Cancellable";

export class KeySignal extends Cancellable {
	constructor(
		/** The keyboard key. */
		public readonly key: Key,

		/** `UIProcessed` is `true` if the key event occurred while a UI object was selected. */
		public readonly uiProcessed: boolean,
	) {
		super();
	}
}
